File: java/azure-storage-blob-v8-perfstress/src/main/java/com/azure/storage/blob/v8/perfstress/core/RandomBlobTest.java
package com.azure.storage.blob.v8.perfstress.core;

import java.net.URISyntaxException;
import java.util.UUID;

import com.azure.perfstress.PerfStressOptions;
import com.microsoft.azure.storage.StorageException;
import com.microsoft.azure.storage.blob.CloudBlockBlob;

public abstract class RandomBlobTest<TOptions extends PerfStressOptions> extends ContainerTest<TOptions> {
    protected final CloudBlockBlob _cloudBlockBlob;
    
    public RandomBlobTest(TOptions options) {
        super(options);

        String blobName = "randomblobtest-" + UUID.randomUUID().toString();

        try {
            _cloudBlockBlob = CloudBlobContainer.getBlockBlobReference(blobName);
        } catch (URISyntaxException | StorageException e) {
            throw new RuntimeException(e);
        }
    }
}

File: net/Azure.Storage.Blobs.V11.PerfStress/Core/RandomBlobTest.cs
using Azure.Storage.Blobs.PerfStress.Core;
using Microsoft.Azure.Storage.Blob;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Storage.Blobs.PerfStress
{
    public abstract class RandomBlobTest<TOptions> : ContainerTest<TOptions> where TOptions: SizeOptions
    {
        protected CloudBlockBlob CloudBlockBlob { get; private set; }

        public RandomBlobTest(TOptions options) : base(options)
        {
            var blobName = "uploadv11test-" + Guid.NewGuid();
            CloudBlockBlob = CloudBlobContainer.GetBlockBlobReference(blobName);
        }
    }
}
